src/components/BCHDExplorer.vue

```diff
@@ -89,32 +89,59 @@ export default {
       getInfoBar: true,
     };
   },
+  mounted() {
+    const params = this.$route.params
+    if (params.address) {
+      this.searchBCHD(params.address)
+    } else if (params.blockHash) {
+      this.searchBCHD(params.blockHash)
+    } else if (params.txId) {
+      this.searchBCHD(params.txId)
+    }
+  },
+  watch: {
+    $route(to) {
+      this.input = ""
+      const input = to.params.address || to.params.blockHash || to.params.txId
```

Comment:
Here we should be able to look up `to.params.network` then make sure the network is set correctly and "tick" then testnet checkbox if necessary.

Reply:
Yep, I had done that in a commit I hadn't pushed until just now.  Will update it to use `to.params.network`.  I spent some time this weekend trying to mock the rpc library but didn't have much success with jest.

Reply:
I had the same problems. We can merge these changes without the mocks. Finding a good testing pattern might take some time.